Koha/Illrequest.pm

```diff
@@ -1346,57 +1330,282 @@ EOF
             "No target email addresses found. Either select at least one partner or check your ILL partner library records.")
           if ( !$to );
         # Create the from, replyto and sender headers
-        my $from = $branch->branchemail;
+        my $from = $branch->branchillemail || $branch->branchemail;
         my $replyto = $branch->branchreplyto || $from;
```

Comment:
This could be changed to reflect the new option for an alternate ILL inbound address.. either `$branch->branchillemail || $branch->branchreplyto || $from` or perhaps use the new `get_staff_to_address` method added below